indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/AbstractParallelIndexSupervisorTaskTest.java

```diff
@@ -546,12 +548,17 @@ public ListenableFuture<Void> runTask(String taskId, Object taskObject)
 
     @Override
     public ListenableFuture<Map<String, Object>> taskReportAsMap(String taskId)
```

Comment:
Can `taskReportAsMap()` now return the concrete type `TaskReport.ReportMap` instead of `Map<String, Object>`?

Reply:
Yes, I have that change in a follow up PR. Didn't do it here as it requires moving all the `TaskReport` related classes to the `druid-processing` module, so that `OverlordClient` can use it.